pkg/gardenlet/controller/seed/seed/reconciler_reconcile.go

```diff
@@ -490,6 +490,24 @@ func (r *Reconciler) runReconcileSeedFlow(
 			},
 			Dependencies: flow.NewTaskIDs(deployCachePrometheus),
 		})
+		deployAlertmanager = g.Add(flow.Task{
+			Name: "Deploying Alertmanager",
+			Fn:   c.alertManager.Deploy,
+		})
+		// TODO(rfranzke): Remove this after v1.92 has been released.
+		_ = g.Add(flow.Task{
+			Name: "Cleaning up legacy Alertmanager resources",
+			Fn: func(ctx context.Context) error {
+				return kubernetesutils.DeleteObjects(ctx, r.SeedClientSet.Client(),
+					&corev1.Service{ObjectMeta: metav1.ObjectMeta{Name: "alertmanager-client", Namespace: r.GardenNamespace}},
+					&corev1.Service{ObjectMeta: metav1.ObjectMeta{Name: "alertmanager", Namespace: r.GardenNamespace}},
+					&appsv1.StatefulSet{ObjectMeta: metav1.ObjectMeta{Name: "alertmanager", Namespace: r.GardenNamespace}},
+					&corev1.Secret{ObjectMeta: metav1.ObjectMeta{Name: "alertmanager-config", Namespace: r.GardenNamespace}},
+					&corev1.PersistentVolumeClaim{ObjectMeta: metav1.ObjectMeta{Name: "alertmanager-db-alertmanager-0", Namespace: r.GardenNamespace}},
```

Comment:
We just delete the old persistent volume of `alertmanager`. Does it not contain any relevant data? Why is it then at all a persistent volume instead of an `emptyDir`? Maybe this is not relevant to this PR, though, and rather a meta question indicating my restricted knowledge about `alertmanager`...

Reply:
Yes, as @oliver-goetz explained, due to the `DataMigration` field in the `alertmanager.Values`, the PV is migrated just like we did it for Prometheus as part of #9128. Deleting the PVC here is not really necessary because the migration code already deletes it, I just listed this here "for completeness" to not miss any resource.

Reply:
Thanks for the explanation. I was confused/surprised when going through the commits in order as at that point in time there was no migration.